Allow editing of person's date of birth and income

LeongWZ opened the pull request:
Closes #89.

This commit allows user to edit income and date of birth fields on `Person` class.
- The user may edit the income field by typing `edit INDEX income/INCOME_VALUE`, e.g. `edit 1 income/900.99`.
- The user may edit the date of birth field by typing `edit INDEX dob/NEW_DATE`, e.g. `edit 1 dob/18 Jun 1999`.
- The user may edit both at the same time with other fields, e.g. `edit 1 income/200 dob/20 Dec 2000 pri/HIGH`.

This commit also updates the message usage in both `EditCommand` and `AddCommand`.

LeongWZ committed:
Merge branch 'master' of github.com:AY2425S1-CS2103T-F14b-1/tp

LeongWZ committed:
Merge branch 'master' of github.com:AY2425S1-CS2103T-F14b-1/tp

LeongWZ committed:
Allow edit of person's date of birth

LeongWZ committed:
Allow edit of person's income

LeongWZ committed:
Update message usage for EditCommand

arjun2598 commented on src/main/java/seedu/address/logic/parser/EditCommandParser.java:
```diff
@@ -46,7 +48,7 @@ public EditCommand parse(String args) throws ParseException {
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
-                PREFIX_REMARK, PREFIX_PRIORITY);
+                PREFIX_REMARK, PREFIX_PRIORITY, PREFIX_DATE_OF_BIRTH);
```
Should the prefix for income also be checked for no duplicates here?

LeongWZ replied:
Thanks for catching later

arjun2598 commented on src/test/java/seedu/address/logic/commands/EditPersonDescriptorTest.java:
```diff
@@ -65,8 +75,10 @@ public void toStringMethod() {
-                + editPersonDescriptor.getRemark().orElse(null) + ", tags="
+                + editPersonDescriptor.getIncome().orElse(null) + ", remark="
+                + editPersonDescriptor.getRemark().orElse(null) + ", dateOfBirth="
+                + editPersonDescriptor.getDateOfBirth().orElse(null) + ", income="
```
Seems like the income and priority strings got swapped when adding in the new fields?

LeongWZ replied:
Noted, the ordering will be corrected in the next commit

arjun2598 reviewed (commented):
LGTM, just some minor clarifications.

Gra7ityIC3 commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -40,8 +40,10 @@ public class AddCommand extends Command {
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
+            + PREFIX_DATE_OF_BIRTH + "9 Mar 1999"
             + PREFIX_PRIORITY + "HIGH "
             + PREFIX_REMARK + "Promises to pay back next week. "
+            + PREFIX_INCOME + "2000"
```
There should be a whitespace after `dob/9 Mar 1999` and `income/2000`.

LeongWZ replied:
Noted. Thanks for catching the mistake

Gra7ityIC3 reviewed (commented):
Aside from some minor changes, everything looks good.

nws321 commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -40,8 +40,10 @@ public class AddCommand extends Command {
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
+            + PREFIX_DATE_OF_BIRTH + "9 Mar 1999"
```
Might want to consider standardising across the board the way user inputs date so that it is the same for dob and scheduling

nws321 reviewed (commented):
LGTM!

LeongWZ committed:
Make input order of Person fields consistent

This commit ensures that the input order of `Person` fields is
consistent through out `edit` and `add` commands.

The order is as follows:
1. Name
2. Phone
3. Email
4. Address
5. Date of birth
6. Priority
7. Income
8. Remark
9. Tags

TerenceWongsExtravaganza reviewed (approved):
LGTM!
